lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -56,6 +59,30 @@ function initializeImportMeta(meta, context, loader) {
   }
 
   meta.url = url;
+  meta.node = {
+    /**
+     * @param {string | URL} specifier
+     * @param {URL['href'] | URL} [parentURL]
+     * @return {Promise<URL>}
+     */
+    async resolveURL(specifier, parentURL = url) {
+      try {
+        const { url: resolved } = await loader.resolve(specifier, parentURL, kEmptyObject);
+        return new URL(resolved);
+      } catch (error) {
+        switch (error?.code) {
+          case 'ERR_UNSUPPORTED_DIR_IMPORT':
+          case 'ERR_MODULE_NOT_FOUND': {
+            const { url: resolved } = error;
+            if (resolved) {
+              return new URL(resolved);
```

Comment:
Does `import.meta.resolve` allow this? I feel like it perhaps should, like it should produce a URL regardless of whether or not the resource exists, but does it? If it’s an intentional variance, and we don’t plan to update `import.meta.resolve` to match, then we should document this difference.